### ISSUE TITLE
WARN: devpi-client-6.0.2 got an unversioned reply, assuming API-VERSION 2

### ISSUE THREAD

**Issue by ralhei** [devpi/devpi#935]
I get the following warning / error when I'm trying to connect to the devpi-server using `devpi use http://localhost:3141`
```
WARN: devpi-client-6.0.2 got an unversioned reply, assuming API-VERSION 2 (as implemented by devpi-server-2.0)
GET http://localhost:3141/+api
502 cannotconnect
```
`devpi-server` got started (inside supervisor) with:
```
devpi-server --serverdir /fs/local/run/devpi-indices
```

The output of devpi running in supervisor is:
```
2022-11-17 20:43:50,936 INFO  NOCTX Loading node info from /fs/local/run/devpi-indices/.nodeinfo
2022-11-17 20:43:50,937 INFO  NOCTX wrote nodeinfo to: /fs/local/run/devpi-indices/.nodeinfo
2022-11-17 20:43:50,955 INFO  NOCTX running with role 'standalone'
2022-11-17 20:43:50,962 WARNI NOCTX No secret file provided, creating a new random secret. Login tokens issued before are invalid. Use --secretfile option to provide a persistent secret. You can create a proper secret with the devpi-gen-secret command.
2022-11-17 20:43:52,420 INFO  [ASYN] Starting asyncio event loop
2022-11-17 20:43:52,430 INFO  NOCTX devpi-server version: 6.7.0
2022-11-17 20:43:52,430 INFO  NOCTX serverdir: /fs/local/run/devpi-indices
2022-11-17 20:43:52,430 INFO  NOCTX uuid: a199e06d59084d74a65d76f2b847af9a
2022-11-17 20:43:52,430 INFO  NOCTX serving at url: http://localhost:3141 (might be http://[localhost]:3141 for IPv6)
2022-11-17 20:43:52,430 INFO  NOCTX using 50 threads
2022-11-17 20:43:52,431 INFO  NOCTX bug tracker: https://github.com/devpi/devpi/issues
2022-11-17 20:43:52,431 INFO  NOCTX Hit Ctrl-C to quit.
2022-11-17 20:43:52,439 INFO  Serving on http://127.0.0.1:3141
```
The server-dir has been initialized with:
```
devpi-init  --serverdir /fs/local/run/devpi-indices --root-passwd '<some-pwd>'
```

I'm running on Python3.9 on a rhel7 machine.

The `pip list` command provides the following output:
```
Package              Version
-------------------- ---------
aiohttp              3.8.3
aiosignal            1.3.1
argon2-cffi          21.3.0
argon2-cffi-bindings 21.2.0
async-timeout        4.0.2
attrs                22.1.0
build                0.9.0
certifi              2022.9.24
cffi                 1.15.1
charset-normalizer   2.1.1
check-manifest       0.48
defusedxml           0.7.1
devpi-client         6.0.2
devpi-common         3.7.0
devpi-server         6.7.0
frozenlist           1.3.3
hupper               1.10.3
idna                 3.4
itsdangerous         2.1.2
lazy                 1.5
multidict            6.0.2
packaging            21.3
passlib              1.7.4
PasteDeploy          3.0.1
pep517               0.13.0
pip                  22.3.1
pkginfo              1.8.3
plaster              1.1
plaster-pastedeploy  1.0.1
platformdirs         2.5.4
pluggy               1.0.0
py                   1.11.0
pycparser            2.21
pyparsing            3.0.9
pyramid              2.0
python-dateutil      2.8.2
repoze.lru           0.7
requests             2.28.1
ruamel.yaml          0.17.21
ruamel.yaml.clib     0.2.7
setuptools           65.3.0
six                  1.16.0
strictyaml           1.6.2
tomli                2.0.1
translationstring    1.4
urllib3              1.26.12
venusian             3.0.0
waitress             2.1.2
WebOb                1.8.7
wheel                0.37.1
yarl                 1.8.1
zope.deprecation     4.4.0
zope.interface       5.5.2
```

**Comment by fschulze** [comment]
Have you tried with http://127.0.0.1:3141? Have you tried either with ``curl`` or a browser?

**Comment by fschulze** [comment]
How are you invoking pip? If you can't connect with devpi, then pip shouldn't work either. Maybe use ``pip -v`` to see details about the used URLs for downloading etc.

**Comment by ralhei** [comment]
Florian - thanks a lot for looking into it. I'm not at work today, let me check again on Monday, and I'll come back to.

**Comment by wesleykendall** [comment]
I also received the same error today when using devpi-server 6.7.0 and devpi-client 6.0.2 for an authenticated server

```
> devpi use http://localhost/
WARN: devpi-client-6.0.2 got an unversioned reply, assuming API-VERSION 2 (as implemented by devpi-server-2.0)
GET http://localhost/+api
401 Unauthorized
```

I can to go to http://localhost just fine and interact with the server, along with installing packages from it. Will try to downgrade versions and see if that matters

**Comment by wesleykendall** [comment]
My issue was that I had upgraded devpi-server to 6 but did not update my nginx config.

After using `devpi-gen-config`, I updated my nginx and it worked again

**Comment by fschulze** [comment]
You can also try ``devpi --debug use http://localhost:3141`` to get more info on the error.

**Comment by ralhei** [comment]
I found what my problem was (a bit embarrassing ;-):  In order to reach the public pypi.org server I had to start devpi-server with http-proxy settings (via env variables). The shell where I used pip to connect to devpi-server did not have these settings (which is correct) and hence worked fine, but the third shell where I tried to connect using the devpi-client had mistakenly also setup http proxy settings, and couldn't connect. Argh. After removing those proxy settings devpi-client works as expected. Sorry for the trouble here.
